dot/core/service.go

```diff
@@ -548,3 +574,41 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	rt.SetContextStorage(ts)
 	return rt.Metadata()
 }
+
+// RegisterRuntimeUpdatedChannel function to register chan that is notified when runtime version changes
+func (s *Service) RegisterRuntimeUpdatedChannel(ch chan<- runtime.Version) (byte, error) {
+	s.runtimeUpdateSubscriptionsLock.Lock()
+	defer s.runtimeUpdateSubscriptionsLock.Unlock()
+
+	if len(s.runtimeUpdateSubscriptions) == 256 {
+		return 0, errors.New("channel limit reached")
+	}
+
+	var id byte
+	for {
+		id = generateID()
+		if s.runtimeUpdateSubscriptions[id] == nil {
+			break
+		}
+	}
+
+	s.runtimeUpdateSubscriptions[id] = ch
+	return id, nil
+}
+
+// UnregisterRuntimeUpdatedChannel function to unregister runtime updated channel
+func (s *Service) UnregisterRuntimeUpdatedChannel(id byte) bool {
+	ch, ok := s.runtimeUpdateSubscriptions[id]
+	if ok {
+		close(ch)
+		s.runtimeUpdateSubscriptions[id] = nil
+		return true
+	}
+	return false
+}
+
+func generateID() byte {
+	// skipcq: GSC-G404
+	id := rand.Intn(256) //nolint
```

Comment:
maybe use a uuid?

Reply:
Updated to use uuid, do I still need to do the loop check since this is a uuid?

Reply:
Yea you should still check even though it's extremely unlikely.